File: gearshift2/src/main/java/org/sugr/gearshift/viewmodel/util/ResourceUtils.java
package org.sugr.gearshift.viewmodel.util;

import android.support.annotation.ArrayRes;

import org.sugr.gearshift.App;

public class ResourceUtils {
    public static int[] stringArrayAsInt(@ArrayRes int stringArrayRes) {
        String[] res = App.get().getResources().getStringArray(stringArrayRes);
        int[] ret = new int[res.length];

        for (int i = 0; i < res.length; i++) {
            ret[i] = Integer.parseInt(res[i]);
        }

        return ret;
    }
}

File: gearshift2/src/main/kotlin/org/sugr/gearshift/viewmodel/util/ResourceUtils.kt
package org.sugr.gearshift.viewmodel.util

import android.support.annotation.ArrayRes

import org.sugr.gearshift.App

object ResourceUtils {
    fun stringArrayAsInt(@ArrayRes stringArrayRes: Int): IntArray {
        val res = App.get().resources.getStringArray(stringArrayRes)
        val ret = IntArray(res.size)

        for (i in res.indices) {
            ret[i] = Integer.parseInt(res[i])
        }

        return ret
    }
}
